mirage/apt/apt_inputs.py

```diff
@@ -338,6 +341,32 @@ def create_input_table(self, verbose=False):
         ascii.write(Table(self.exposure_tab), self.output_csv, format='csv', overwrite=True)
         print('csv exposure list written to {}'.format(self.output_csv))
 
+
+    def check_aperture_override(self):
+        if bool(self.exposure_tab['FiducialPointOverride']) == True:
+            instruments = self.exposure_tab['Instrument']
+            apertures = self.exposure_tab['aperture']
+
+            aperture_key = constants.instrument_abbreviations
+
+            fixed_apertures = []
+            for i, (instrument, aperture) in enumerate(zip(instruments, apertures)):
+                inst_match_ap = aperture.startswith(aperture_key[instrument.lower()])
+                if not inst_match_ap:
+                    # Handle the one case we understand, for now
+                    if instrument.lower() == 'fgs' and aperture[:3] == 'NRC':
+                        obs_num = self.exposure_tab['obs_num'][i]
+                        guider_number = read_apt_xml.get_guider_number(self.input_xml, obs_num)
+                        guider_aperture = 'FGS{}_FULL'.format(guider_number)
+                        fixed_apertures.append(guider_aperture)
+                    else:
+                        raise ValueError('Unknown FiducialPointOverride in program. Instrument = {} but aperture = {}.'.format(instrument, aperture))
+                else:
+                    fixed_apertures.append(aperture)
+
+            self.exposure_tab['aperture'] = fixed_apertures
```

Comment:
Could we in addition define a new key `pointing_aperture` to retain all the information? e.g.
```
self.exposure_tab['pointing_aperture'] = self.exposure_tab['aperture']
self.exposure_tab['aperture'] = fixed_apertures
```
I am not sure if this will satisfy all cases but we will have an available record of the aperture that is specified in the pointing file.

Reply:
I added this in the way you described here. One thought is that it's within the if statement that checks if a given program has an override, and so the `self.exposure_tab['pointing_aperture']` key will only be available if `self.exposure_tab['FiducialPointOverride']` is True.

Reply:
Yes, good catch. That is not good. `self.exposure_tab['pointing_aperture']` should always be populated.

Reply:
Okay, this should be fixed now!